src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js

```diff
@@ -46,8 +49,11 @@ export class DeviceDelete extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.devices && (this.props.devices || []).length !== nextProps.devices.length) {
-      this.populateDevicesState(nextProps.devices);
+    const nextDevices = this.ensureDevicesExist(nextProps.devices);
+    const curDevices = this.state.devices || [];
```

Comment:
Isn't `state.devices` guaranteed to always be initialized to an array (as on line 34)?